sidecar-ffi/src/lib.rs

```diff
@@ -709,3 +710,18 @@ pub extern "C" fn ddog_sidecar_reconnect(
 ) {
     transport.reconnect(|| unsafe { factory() });
 }
+
+/// Return the path of the crashtracker unix domain socket.
+#[no_mangle]
+#[allow(clippy::missing_safety_doc)]
+pub unsafe extern "C" fn ddog_sidecar_get_crashtracker_unix_socket_path() -> ffi::CharSlice<'static>
+{
+    let socket_path = crashtracker_unix_socket_path();
+    let str = socket_path.to_str().unwrap_or_default();
+
+    let size = str.len();
```

Comment:
Would it be cleaner to take a trip through https://doc.rust-lang.org/std/primitive.str.html#method.into_boxed_bytes

Reply:
At the cost of not being able to free() it in C code. This impl allows avoiding to need extra drop routines / relying on the specific allocator rust currently uses.